src/FTPClientWrapperSSH.cpp

```diff
@@ -553,47 +552,19 @@ int FTPClientWrapperSSH::authenticate_key(ssh_session session) {
 	}
 
 	char * keyfile = SU::TCharToCP(m_keyFile, CP_ACP);
-	ssh_private_key privkey = NULL;
-	if (m_passphrase[0] == 0)	//in case the passphrase is empty, use NULL instead
-		privkey = privatekey_from_file(session, keyfile, 0, NULL);
-	else
-		privkey = privatekey_from_file(session, keyfile, 0, m_passphrase);
+	ssh_key privkey = NULL;
+	//in case the passphrase is empty, use NULL instead
+	rc = ssh_pki_import_privkey_file(keyfile, m_passphrase[0] ? m_passphrase : NULL, NULL, NULL, &privkey);

```

Comment:
@ashkulz What do you think? With or without additional file existance check? The change seems ok to me, according to http://api.libssh.org/stable/libssh_tutor_authentication.html.

Might be even changed to ssh_userauth_publickey_auto() and that one is trying to do all magically.